tests/solana_tests/optimizations.rs

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    borsh_encoding::{visit_mut, VisitorMut},
+    BorshToken, VirtualMachineBuilder,
+};
+use anchor_syn::idl::IdlInstruction;
+use once_cell::sync::Lazy;
+use serde::Deserialize;
+use solang::codegen::Options;
+use std::{
+    env::var,
+    fs::{read_dir, read_to_string, File},
+    io::BufReader,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Deserialize)]
+struct Calls {
+    constructor: Vec<BorshToken>,
+    function: Vec<(String, Vec<BorshToken>)>,
+}
+
+static NO_OPTIMIZATIONS: Lazy<Options> = Lazy::new(|| Options {
+    dead_storage: false,
+    constant_folding: false,
+    strength_reduce: false,
+    vector_to_slice: false,
+    common_subexpression_elimination: false,
+    ..Default::default()
+});
+
+#[test]
+fn optimizations() {
+    let calls = Path::new("tests/optimization_testcases/calls");
+
+    if let Ok(testname) = var("TESTNAME") {
+        run_tests(std::iter::once(calls.join(testname).with_extension("json")));
+    } else {
+        run_tests(read_dir(calls).unwrap().map(|entry| entry.unwrap().path()));
+    }
+}
+
+fn run_tests(iter: impl Iterator<Item = PathBuf>) {
+    for path in iter {
+        let file_stem = path.file_stem().unwrap();
+
+        // Known problematic test.
+        if file_stem == "a467d917a5158b613f94954d4a558a5cb61e6773" {
+            continue;
+        }
+
+        dbg!(file_stem);
+
+        let file = File::open(&path).unwrap();
```

Comment:
This should be an `expect(...)` explaining why this can never fail, otherwise print a more friendly error message (when this test panics here it'll be a very annoying "Os Error" on the open which is super annoying to debug).

Reply:
I changed this to `unwrap_or_else(|error| panic!(...))`.

Reply:
perfect, thanks